src/resource_manager/gateway.py

```diff
@@ -62,29 +57,40 @@ def _label_selector(self) -> str:
         return ",".join(f"{k}={v}" for k, v in self._labels.items())
 
     @map_k8s_auth_exception
-    def _gen_resource(self, definition: GatewayResourceDefinition, config: CharmConfig) -> dict:
+    def _gen_resource(self, state: State) -> dict:
         """Generate a Gateway resource from a gateway resource definition.
 
         Args:
-            definition: The gateway resoucre definition to use.
-            config: The charm's configuration.
+            state: Part of charm state consisting of 3 components:
+                - GatewayResourceManager
+                - CharmConfig
+                - SecretResourceManager
 
         Returns:
             A dictionary representing the gateway custom resource.
         """
+        tls_secret_name = f"{state.secret_resource_name_prefix}-{state.external_hostname}"
         gateway = self._gateway_generic_resource(
             apiVersion="gateway.networking.k8s.io/v1",
             kind="Gateway",
-            metadata=ObjectMeta(name=definition.gateway_name, labels=self._labels),
+            metadata=ObjectMeta(name=state.gateway_name, labels=self._labels),
             spec={
-                "gatewayClassName": config.gateway_class,
+                "gatewayClassName": state.gateway_class,
```

Comment:
Just out of curiosity, could this spec dict be typed?

Reply:
Good point, I think I looked into this when I started with the gateway implementation. I just looked again and it seems like there's no typed spec for `GenericNamespacedResource`